### PR TITLE
Fixing issue #296 "Inconsistent requirement of course id"

### PR DESCRIPTION
**Pull request by lphk92**

### CODE REVIEW

File: nbgrader/apps/collectapp.py [hunk 1]
```diff
@@ -73,12 +73,9 @@ class CollectApp(TransferApp):
...
         help="Update existing submissions with ones that have newer timestamps."
     )
-    
+
     def init_args(self):
-        if len(self.extra_args) == 1:
-            self.assignment_id = self.extra_args[0]
-        else:
-            self.fail("Invalid number of argument, call as `nbgrader release ASSIGNMENT`.")
+        pass
```
Comment (jhamrick) [inline nbgrader/apps/collectapp.py]: I think this function can just be removed.

File: nbgrader/apps/fetchapp.py [hunk 2]
```diff
@@ -24,33 +24,29 @@ class FetchApp(TransferApp):
...
         """
-    
+
     def init_args(self):
-        if len(self.extra_args) == 2:
-            self.course_id = self.extra_args[0]
-            self.assignment_id = self.extra_args[1]
-        else:
-            self.fail("Invalid number of args, call as `nbgrader fetch COURSE ASSIGNMENT`")
+        pass
```
Comment (jhamrick) [inline nbgrader/apps/fetchapp.py]: This can be removed.

File: nbgrader/apps/listapp.py [hunk 3]
```diff
@@ -36,64 +36,59 @@ class ListApp(TransferApp):
...
-    
+
     def init_args(self):
-        if len(self.extra_args) == 0:
-            pass
-        elif len(self.extra_args) == 1:
-            self.course_id = self.extra_args[0]
-        else:
-            self.fail("Invalid number of argument, call as `nbgrader list COURSE`.")
+        pass
```
Comment (jhamrick) [inline nbgrader/apps/listapp.py]: This can be removed.

File: nbgrader/apps/releaseapp.py [hunk 4]
```diff
@@ -37,44 +37,41 @@ class ReleaseApp(TransferApp):
...
         """
 
     force = Bool(False, config=True, help="Force overwrite existing files in the exchange.")
 
     def init_args(self):
-        if len(self.extra_args) == 1:
-            self.assignment_id = self.extra_args[0]
-        else:
-            self.fail("Invalid number of argument, call as `nbgrader release ASSIGNMENT`.")
+        pass
```
Comment (jhamrick) [inline nbgrader/apps/releaseapp.py]: This can be removed.

File: nbgrader/apps/submitapp.py [hunk 5]
```diff
@@ -30,38 +30,34 @@ class SubmitApp(TransferApp):
...
         """
-    
+
     def init_args(self):
-        if len(self.extra_args) == 2:
-            # The first argument (assignment_id) is processed in init_src
-            self.course_id = self.extra_args[1]
-        else:
-            self.fail("Invalid number of args, call as `nbgrader submit assignment_id course_id`")
+        pass
```
Comment (jhamrick) [inline nbgrader/apps/submitapp.py]: This can be removed.

File: nbgrader/apps/baseapp.py [hunk 6]
```diff
@@ -259,13 +263,13 @@ def initialize(self, argv=None):
     def init_src(self):
         """Compute and check the source paths for the transfer."""
         raise NotImplemented
-    
+
```
Comment (jhamrick) [inline nbgrader/apps/baseapp.py]: I can't comment on the relevant line, but I think the `init_args` in this class can also be removed and the call to it in the `initialize` function.

File: nbgrader/apps/fetchapp.py [hunk 7]
```diff
@@ -24,33 +24,29 @@ class FetchApp(TransferApp):
...
+
             nbgrader list phys101
-        
+
         To fetch an assignment by name into the current directory:
-        
+
             nbgrader fetch phys101 assignment1
-        
+
```
Comment (jhamrick) [inline nbgrader/apps/fetchapp.py]: This documentation needs to be updated.

File: nbgrader/apps/listapp.py [hunk 8]
```diff
@@ -36,64 +36,59 @@ class ListApp(TransferApp):
...
+
         To see all of the released assignments, run
-        
+
             nbgrader list  # course_id in the config file
-        
+
         or
-        
+
```
Comment (jhamrick) [inline nbgrader/apps/listapp.py]: This documentation needs to be updated to use the `--course` flag.

File: nbgrader/apps/submitapp.py [hunk 9]
```diff
@@ -30,38 +30,34 @@ class SubmitApp(TransferApp):
...
         `assignment1`, that must be a sub-directory of your current working directory.
         If you are inside the `assignment1` directory, it won't work.
-        
+
         To fetch an assignment you must first know the `course_id` for your course.
         If you don't know it, ask your instructor.
 
         To submit `assignment1` to the course `phys101`:
-        
+
```
Comment (jhamrick) [inline nbgrader/apps/submitapp.py]: This documentation needs to be updated.

**Comment by jhamrick** [comment]
This looks great, thanks! I think there's just a few small changes that need to be made. I also think probably some of the main documentation (in nbgrader/docs/source/user_guide) needs to be updated since the commands have changed.

**Comment by jhamrick** [comment]
Aw, damn, these tests are going to fail because I forgot to tell you about `clear_docs` -- basically, when you save a notebook, it will put some metadata about the kernel into the json which then will make the notebook look for either python 2 or python 3 -- but the notebooks should be able to work with either python 2 or python 3, so you need to remove the metadata before committing. If you run `invoke clear_docs` from the root of the repo it will do that for you and then you can recommit.

**Comment by lphk92** [comment]
I just ran `clear_docs` and submitted, so hopefully that'll make Travis happy!

**Comment by jhamrick** [comment]
Yay, it passes! Thanks!

**Pull request merged**